### PR TITLE
adding tests and pulling gdp helper functions out

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
I'll be adding to this, but just opening so you can see what I'm up to. Don't merge yet.

### CODE REVIEW

**Comment by jread-usgs** [comment]
@dblodgett-usgs what is missing from `geoknife` that you need here? It probably doesn't make a ton of sense to have gdp-specific functions in this package, right?

File: R/gdp_ids_and_units.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#'
+#'@param gdp_file A Geo Data Portal output file containing one statistic, currently no support for multiple statistic output files.
+#'
+#'@return A named list as follows:\cr
+#'nc_stations contains unique station ids,\cr
+#'nc_data_var_unique_name contains the variables in the file,\cr
+#'unique_units contains the units associated with each variable.
+#'
+#'@export
+gdp_ids_and_units<-function(GDPFile)
```
Comment (jread-usgs) [inline R/gdp_ids_and_units.R]: see `geoknife::parseTimeseries(..., with.units = TRUE)` or `geoknife::loadOutput()`?

Comment (dblodgett-usgs) [inline R/gdp_ids_and_units.R]: Yeah... I'm working through that now. another commit coming that will demonstrate what I've learned.

Comment (dblodgett-usgs) [inline R/gdp_ids_and_units.R]: @lawinslow had dumped my 'gdp helper' functions into the repo so I just started working them into the package without thinking too much. I've gotta spend a lot more time comparing this and geoknife functionality to what is really needed to get them into DSG. This is going to be more work than I thought. See my latest commit reconciling things a bit.

File: NAMESPACE [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # Generated by roxygen2 (4.1.1): do not edit by hand
 
+export(gdp_count_time_steps)
+export(gdp_ids_and_units)
 export(write_point_dsf)
-export(write_timeseries_dsf)
+export(write_timeseries_dsg)
```
Comment (dblodgett-usgs) [inline NAMESPACE]: Discrete Sampling Geometries is the name of this class of things in the NetCDF-CF spec.

File: R/gdp_count_time_steps.R [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#'@title Get Geo Data Portal time step counter.
```
Comment (dblodgett-usgs) [inline R/gdp_count_time_steps.R]: I'm not sure if this should be in this package, but for really big GDP files, it's nice to have something that just looks through the file and pulls out the time steps so you can parse out the data more carefully later and have a fixed size array to write into... I'm not attached to this though.

**Comment by dblodgett-usgs** [comment]
Starting to sink my teeth into this, the test I started for the timeseries DSG function is a good start I think. Should factor out the dependency on geoknife and just load up some canned data or something.

I'll start a list of issues that need to be addressed in this function, not sure what the best implementation is going to be on them, so I'll keep them as functional and not implementation specific as possible. Please don't merge this PR, just a conversation piece for the time being.

File: R/gdp_ids_and_units.R (hunk 1 above)
Comment (jread-usgs) [inline R/gdp_ids_and_units.R]: I see. Your call, but probably want to think about the scope of this package and what should be in it vs elsewhere.

**Comment by dblodgett-usgs** [comment]
Darn it... I totally forgot to sync up my local repo and was working off old code. :-1: Will reconcile with the latest, hopefully not a lot of rework.

**Pull request closed without merging**